webapp/routes.go

```diff
@@ -13,24 +13,27 @@ import (
 var templates = template.Must(template.ParseGlob("templates/*.html"))
 
 func init() {
-	// Test run results, viewed by browser (default view)
-	// For run results diff view, 'before' and 'after' params can be given.
-	shared.AddRoute("/", testResultsHandler)
-	shared.AddRoute("/results", testResultsHandler) // Prevent default redirect
-	shared.AddRoute("/results/", testResultsHandler)
-
 	// About wpt.fyi
-	shared.AddRoute("/about", aboutHandler)
-
-	// Test run results, viewed by pass-rate across the browsers
-	shared.AddRoute("/interop/", interopHandler)
+	shared.AddRoute("/about", "about", aboutHandler)
 
 	// Lists of test run results which have poor interoperability
-	shared.AddRoute("/interop/anomalies", anomalyHandler)
+	shared.AddRoute("/anomalies", "anomaly", anomalyHandler)
+
+	// Test run results, viewed by pass-rate across the browsers
+	shared.AddRoute("/interop/", "interop", interopHandler)
+	shared.AddRoute("/interop/{path}", "interop", interopHandler)
 
 	// List of all test runs, by SHA[0:10]
-	shared.AddRoute("/test-runs", testRunsHandler)
+	shared.AddRoute("/test-runs", "test-runs", testRunsHandler)
 
 	// Admin-only manual results upload.
-	shared.AddRoute("/admin/results/upload", adminUploadHandler)
+	shared.AddRoute("/admin/results/upload", "upload", adminUploadHandler)
+
+	// Test run results, viewed by browser (default view)
+	// For run results diff view, 'before' and 'after' params can be given.
+	shared.AddRoute("/results/{path}", "results", testResultsHandler)
+
+	// Legacy wildcard match
+	shared.AddRoute("/", "results-legacy", testResultsHandler)
+	shared.AddRoute("/{path}", "results-legacy", testResultsHandler)
```

Comment:
Could you explain to me (or point me to a doc) how Gorilla mux handles multiple matches? For example here, `"/{path}"` can also match `/interop`, `/test-runs`, etc. Is there a priority?

Reply:
https://github.com/gorilla/mux#matching-routes

> Routes are tested in the order they were added to the router. If two routes match, the first one wins

Reply:
Thanks for the pointer!

Since we set up routes in `init()`, how do we know for sure which `init()` runs first? For example, if `webapp/routes.go` sets up before `api/routes.go`, would this catch-all rule render all the routes in `api/routes.go` useless?

Reply:
Good point. I think this has been a potential issue the whole time (not just before gorilla/mux), so I've made the calling order explicit during the `init()` call of `webapp`